tests/test_snapshots.py

```
@@ -0,0 +1,1530 @@
+        side_effect=exc,
+    ):
+        result = downloader.fetch_snapshot_release()
+    assert result is None
+
+
+# ------------------------------------------------------------------
+# Integration: end-to-end decision flow
+# ------------------------------------------------------------------
+
+
+def test_full_snapshot_decision_flow(downloader, cache_manager):
```

Ruff (0.15.20)

[warning] Unused function argument: `cache_manager`

(ARG001)